### ISSUE TITLE
libmono-native.so size has increased by 643KB in VS 16.3.2

### ISSUE THREAD

**Issue by bellissimo** [xamarin/xamarin-android#3799]
Since VS 16.3.2 I believe, I noticed that my APK size had increased by about half a megabyte without any significant code changes being made. Upon further investigation I believe that a large portion of this is down to the size of libmono-native.so increasing substantially. It has gone from 22KB to 665KB.

I did notice that several other system libraries have also increased in size, though none so dramatically as this. 

I am raising this to establish whether this is a genuinely required increase in size, or whether there is possibly some issue here (e.g. with linking possibly?). 

At a time when Google are encouraging us to reduce our app size, adding half a megabyte to the APK is not an insignificant change.

### Steps to Reproduce

Create a release build for an Android app
AOT and LLVM both turned off

### Behavior

libmono-native.so is significantly larger with VS 16.3.2 than it was before

### Version Information

Microsoft Visual Studio Professional 2019
Version 16.3.3
VisualStudio.16.Release/16.3.3+29403.142
Microsoft .NET Framework
Version 4.8.03752

Installed Version: Professional

Visual C++ 2019   00435-60000-00000-AA950
Microsoft Visual C++ 2019

Application Insights Tools for Visual Studio Package   9.1.00913.1
Application Insights Tools for Visual Studio

Azure App Service Tools v3.0.0   16.3.284.37798
Azure App Service Tools v3.0.0

C# Tools   3.3.1-beta3-19461-02+2fd12c210e22f7d6245805c60340f6a34af6875b
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.0 (d16-2@8b56e20)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.3.7 (9d260c5)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Visual Basic Tools   3.3.1-beta3-19461-02+2fd12c210e22f7d6245805c60340f6a34af6875b
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.4 for F# 4.6   16.3.0-beta.19455.1+0422ff293bb2cc722fe5021b85ef50378a9af823
Microsoft Visual F# Tools 10.4 for F# 4.6

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.3.0.275 (d16-3@35ef585)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.3.0.246 (remotes/origin/d16-3@bd2f86892)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.3.565 (27e9746)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   10.0.3.0 (d16-3/4d45b41)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: mono/mono/2019-06@5608fe0abb3
    Java.Interop: xamarin/java.interop/d16-3@5836f58
    LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
    LibZip: nih-at/libzip/rel-1-5-1@b95cf3fd
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.27.1@8212a2d
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333

Xamarin.iOS and Xamarin.Mac SDK   13.2.0.47 (c2cbd34)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

**Comment by dellis1972** [comment]
@akoeplinger is this and expected size increase?

**Comment by akoeplinger** [comment]
@dellis1972 yes this is expected because of the new Brotli encoder support which is required by .NET Standard: https://github.com/mono/mono/pull/12189#issuecomment-453650140

**Comment by bellissimo** [comment]
My app seems to be slightly slower after upgrading to 16.3.2, so I thought I would downgrade to 16.3.1 in case this was caused by the changes in the libraries as discussed above. However after following the instructions here: https://docs.microsoft.com/en-us/visualstudio/releases/2019/history and doing a full cleand and re-build using 16.3.1, the libmono-native.so is still 665KB. I tried downgrading again to 16.3.0, but the same result.

Does downgrading the VS version not also downgrade the mono libraries, or is there something else I need to do to get a clean rollback and the old libmono-native.so?
